Filter Conferences by Date Range

JuanPabloDiaz opened the pull request:
Reference #749
[react-datepicker](https://www.npmjs.com/package/react-datepicker
)
![image](https://github.com/user-attachments/assets/54a710e3-4652-4fa6-be7f-22049170dd37)


![image](https://github.com/user-attachments/assets/2d286434-bf4c-4b0a-9cf0-e735f3d9b7de)

cgrail commented:
Looks good. But there is a compile error

JuanPabloDiaz commented:
> Looks good. But there is a compile error

I finally got it lol

JuanPabloDiaz commented:
![image](https://github.com/user-attachments/assets/1da2a6aa-76d9-44f7-bfda-4eee708c3420)

nimzco commented:
Hey @JuanPabloDiaz, thank you for this proposal and this PR!

In terms of usability, how about only showing months instead of full dates in the picker? I'm even thinking, only showing one month, to be like: "show me conference after this date".

If we were to keep start and end date, I would like the filter to update as soon as I select the start date. Meaning: I don't need to select the end date to see the conference list change.
Additionally, we need to handle the mobile view, or to hide it on mobile.

nimzco commented on src/scenes/ConferenceList/ConferenceList.tsx:
```diff
@@ -84,6 +87,20 @@ const ConferenceListPage: React.FC<Props> = ({
+
+  const handleStartDateSelect = (date: Date | null) => {
+    if (date) {
+      setStartDate(date)
```
Don't you want to handle when the user empties the input?

nimzco commented on src/scenes/ConferenceList/ConferenceList.module.scss:
```diff
@@ -88,3 +88,16 @@
+.dateRangePicker {
+  display: flex;
+  width: 400px;
+  padding: 2px;
```
This padding breaks the vertical alignment, see:
<img width="686" alt="image" src="https://github.com/user-attachments/assets/05a7805c-3fa6-4bfa-a4f6-749807638e4e">

JuanPabloDiaz replied:
wow, I had to zoom in to see that.

Great catch! I will work on it

nimzco commented on src/scenes/ConferenceList/ConferenceList.module.scss:
```diff
@@ -88,3 +88,16 @@
+  display: flex;
+  width: 400px;
+  padding: 2px;
+  gap: 2px;
```
```suggestion
  gap: spacing(tight);
```

nimzco reviewed (changes requested):
A few comments, but it looks good!

When selecting a start date, I would expect the results to change. They don't until we select the end date.
We could also auto-open the end-date after selecting the start-date, but I would start filtering already.

UI:
- The height of the inputs are not the same with the search
- The X is not center-aligned.
<img width="336" height="74" alt="image" src="https://github.com/user-attachments/assets/30f71b24-c510-4bb2-a9c6-7efbf2674e78" />

JuanPabloDiaz commented:
Thanks for your feedback @nimzco! I've implemented your suggestions, and everything should now be working as you requested.

https://github.com/user-attachments/assets/35cfabe8-56b4-440d-a747-7f356e89a80c


feat: improve date filtering UX and fix input alignment

- Filters now apply immediately upon selecting start or end dates.
- Auto-focus end date picker for better UX  
- Fix alignment between search box and date inputs
- Enhance date range overlap detection
- Improve dark mode and accessibility

nimzco commented:
When testing locally, the conference list don't update anymore after I change the dates.

JuanPabloDiaz commented:
Thanks @nimzco! You're absolutely right. 

Date filtering was client-side while other filters were server-side, causing conflicts.

Fixed! Moved date filtering to Algolia level so it works with other filters.

Should be working now! 🎯

nimzco commented on src/scenes/ConferenceList/components/ConferenceList/ConferenceList.module.scss:
```diff
@@ -46,3 +46,21 @@
   margin: spacing.spacing(extra-loose);
 }
+
+.dateRangePicker {
+  display: flex !important;
+  flex-direction: row !important; // Force it to be in a row
+  gap: 0.5rem !important;
+  align-items: center !important;
+  padding: 0.5rem !important;
+
+  .customDatePicker {
+    border: 1px solid #d1d5db !important;
+    padding: 0.25rem !important;
+    width: auto !important;
+    max-width: 12rem !important;
+    height: 35px !important;
+    color: black !important;
+    border-radius: 0.375rem !important;
+  }
+}
```
Are you sure all of this is needed? I don't see the CSS applied and if I remove the code, the behavior is still the same. 🤔

JuanPabloDiaz replied:
you are right, I forgot to remove that. I add it in this [commit](https://github.com/tech-conferences/confs.tech/pull/761/commits/b7f7af39136b2aad6fd8db91c1fb62c2007136a7) but we no longer need it.

nimzco reviewed (approved):
Aside of my last comment. I've tested locally, and it works as expected. 🚢 
Thanks for the feature addition!
